lib/AST/LocalizationFormat.cpp

```diff
@@ -76,17 +73,38 @@ template <> struct MappingTraits<DiagnosticNode> {
 namespace swift {
 namespace diag {
 
-void SerializedLocalizationWriter::add(uint32_t id, llvm::StringRef message) {
-  generator.insert(diagnosticID[id], message);
+void SerializedLocalizationWriter::insert(llvm::StringRef id,
+                                          llvm::StringRef translation) {
+  generator.insert(id, translation);
 }
 
-SerializedLocalizationWriter::~SerializedLocalizationWriter() {
-  uint32_t offset = generator.Emit(OS);
+void SerializedLocalizationWriter::emit(llvm::StringRef filePath) {
+  std::error_code error;
+  llvm::raw_fd_ostream OS(filePath, error, llvm::sys::fs::F_None);
+  uint32_t offset;
+  {
+    llvm::support::endian::write<uint32_t>(OS, 0, llvm::support::little);
+    offset = generator.Emit(OS);
+  }
   OS.seek(0);
   llvm::support::endian::write(OS, offset, llvm::support::little);
   OS.close();
 }
 
+SerializedLocalizationProducer::SerializedLocalizationProducer(
+    std::unique_ptr<llvm::MemoryBuffer> &buffer)
```

Comment:
Since I'm passing a `unique_ptr`. I can't pass it by value (make a copy). Therefore, I passed it by reference. Does that make sense?